pages/api/send-analytics.ts

```diff
@@ -7,7 +7,7 @@ const { serverRuntimeConfig } = getConfig();
 
 export default async function sendAnalytics(req: NextApiRequest, res: NextApiResponse) {
     const {event, data = {}} = req?.body;
-    const anonymousId = req.cookies['helio_analytics_id'] || moment().unix().toString()
+    const anonymousId = req.cookies['user_id'] || req.cookies['helio_analytics_id'] || moment().unix().toString()
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change introduces a potential security issue. The `anonymousId` is now being set to the value of the 'user_id' cookie if it exists, which could lead to user tracking issues if the 'user_id' cookie contains sensitive information. It's generally not a good practice to use potentially identifiable information as an anonymous ID.
```diff
- const anonymousId = req.cookies['user_id'] || req.cookies['helio_analytics_id'] || moment().unix().toString()
+ const anonymousId = req.cookies['helio_analytics_id'] || moment().unix().toString()
```
Also, there might be a logical error. If the 'user_id' cookie is not available, the code falls back to using the 'helio_analytics_id' cookie. However, if that is also not available, it uses the current Unix timestamp as the anonymous ID. This means that every request without these cookies will have a different anonymous ID, which defeats the purpose of having an anonymous ID for tracking purposes. Consider generating a unique but consistent identifier instead.


<!-- This is an auto-generated comment by OSS CodeRabbit -->